### DISCUSSION TITLE
how-to-make-cutouts-in-jetpack-compose-boxes/

### DISCUSSION THREAD

**Discussion by giscus** [JakeSteam/blog-programming#53] [Comments]
# how-to-make-cutouts-in-jetpack-compose-boxes/

At work recently we needed to have a semi-circular “cutout” from a rectangular box containing content. This was surprisingly tricky, so here’s an easy way to cut out arbitrary shapes from a Jetpack Compose Box!

https://blog.jakelee.co.uk/how-to-make-cutouts-in-jetpack-compose-boxes/

**Comment by arenshaee** [comment]
You saved me :)

**Comment by tprochazka** [comment]
It is possible make that text transparent?

**Reply by JakeSteam** [reply]
Good question! I haven't tried, but I don't think it will work with this technique since the text is part of `drawContent`, not the overlayed shapes being made transparent.

There might be techniques in [Android's `GraphicsModifiersSnippets`](https://github.com/android/snippets/blob/f595a0d317fd8aca6b33d4344bf87696cd45c481/compose/snippets/src/main/java/com/example/compose/snippets/graphics/GraphicsModifiersSnippets.kt) or [Custom Drawing Modifiers](https://developer.android.com/reference/kotlin/androidx/compose/ui/draw/package-summary#custom-drawing-modifier).

Good luck, I'd be interested to see if you manage to get it working!
